### ISSUE TITLE
Wrong Type Narrowing of Class Instance with A Member of typing.Literal

### ISSUE THREAD

**Issue by Congee** [microsoft/pyright#1578]
**Describe the bug**
The instance is inferred as Never in if branches that check its member typed `typing.Literal[...]`

**Screenshots or Code**
```python
import typing
class C:
    a: typing.Literal[1,2,3]

c = C()

if c.a == 1:
    ...
elif c.a ==2:  # c is inferred as Never here
    ...
```

**VS Code extension or command-line**
pyright 1.1.117 command line

**Additional context**
It happened since long ago, at least from 1.1.112

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.118, which I just published. It will also be included in the next release of Pylance.
